plugins/tweak/tweaks/stable-cursor.h

```diff
@@ -32,6 +34,14 @@ struct stable_cursor_hook : df::viewscreen_dwarfmodest
         }
     }
 
+    bool check_viewport_near_enough()
+    {
+        df::coord view = Gui::getViewportPos();
+        return abs(view.x - last_view.x) <= near_enough_distance &&
+            abs(view.y - last_view.y) <= near_enough_distance &&
+            view.z == last_view.z;
```

Comment:
should the z-level matter? Wouldn't it still be valuable to keep the cursor position even if the z-level has changed?

Reply:
In my view it's not helpful; on another z level, what the cursor was referencing is not visible, so there's no reason to place the cursor there. Maybe for people who order their fort with one workshop directly over another or something it would be desirable? I could see it either way I guess.

Reply:
Yeah, I could go either way as well. If keeping the cursor on a building (or any specific tile) is the only purpose, then yeah changing z-levels "dirties" the cursor. If the user is expecting the cursor to appear on the same part of the screen when moving z-levels, then it might be unexpected that the cursor jumps back to the middle of the screen.

That said, before I thought to look at this carefully, I never noticed what the stable-cursor tweak did in this situation. How often does a user go into a cursor mode, exit the cursor mode, switch z-levels, and re-enter a cursor mode? Probably not often. If they're switching z-levels with a cursor, they're probably going to stay in the same cursor-having mode while they switch z-levels. tl;dr what you have is fine : )